### PR TITLE
Bug #13812: fixing technical problem

### PR DESCRIPTION
**Pull request by SilverYoCha**
Fixing also:
* Bug 13815
* Bug 13816
* Bug 13817 
* Bug 13818
* Bug 13819

### CODE REVIEW

File: core-web/src/main/java/org/silverpeas/core/web/mvc/controller/AbstractAdminComponentSessionController.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+                      .map(getOrganisationController()::getSpaceInstById)
+                      .filter(Objects::nonNull)
+                      .map(SpaceInst::getDomainFatherId)
+                      .filter(StringUtil::isDefined)
+                      .filter(not(userManageableSpaceIdSet::contains))
+                      .flatMap(m -> getOrganisationController().getPathToSpace(m).stream())
+                      .map(SpaceInstLight::getId)
+                      .anyMatch(s::equals)))
+              .orElse(false);
+        }
```
Comment (mmoqui) [inline core-web/src/main/java/org/silverpeas/core/web/mvc/controller/AbstractAdminComponentSessionController.java:172]: Mmm... I require some explanations here. What I understand from the above code:
access is granted if:
- the space is among the spaces the user is able to manage
- or the access is read-only (not within a side-effect operation) and the parent of the space isn't among the spaces the user is able to manage but there is at least one parent to this parent that is manageable by the user.
What I don't understand is why the parent of the current space to check must not be among the the manageable spaces.

**Review by mmoqui — CHANGES_REQUESTED** [review]

File: core-war/src/main/java/org/silverpeas/web/jobdomain/control/JobDomainPeasSessionController.java [hunk 2]
```diff
@@ -187,11 +197,95 @@ public boolean isUserAddingAllowedForGroupManager() {
...
   public boolean isAccessGranted() {
     return getUserDetail().isAccessAdmin() || getUserDetail().isAccessDomainManager() ||
         isOnlySpaceManager() || !getUserManageableGroupIds().isEmpty();
   }
 
+  public void checkDomainAccessGranted(final String domainId) {
+    checkDomainAccessGranted(domainId, true);
+  }
+
+  public void checkCurrentDomainDomainAccessGranted(final boolean readOnly) {
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-war/src/main/java/org/silverpeas/web/jobdomain/control/JobDomainPeasSessionController.java:210]: That is a mistake.

File: core-war/src/main/java/org/silverpeas/web/jobdomain/control/JobDomainPeasSessionController.java [hunk 3]
```diff
@@ -187,11 +197,95 @@ public boolean isUserAddingAllowedForGroupManager() {
...
+      final boolean readOnly) {
+    final UserDetail ud = getUserDetail();
+    final boolean granted = ud.isAccessAdmin() || isAccessGrantedOnNotFillAdminAccess(
+        defaultStringIfNotDefined(domainId, Domain.MIXED_DOMAIN_ID), accessContext, readOnly);
+    if (!granted) {
+      throwForbiddenError();
+    }
+  }
+
+  private <T> boolean isAccessGrantedOnNotFillAdminAccess(final String domainId,
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-war/src/main/java/org/silverpeas/web/jobdomain/control/JobDomainPeasSessionController.java:254]: This is a mistake. 'NotFullAdminAccess' is meant here.

File: core-web/src/main/java/org/silverpeas/core/web/mvc/controller/AbstractAdminComponentSessionController.java (hunk 1 above)
Comment (SilverYoCha) [reply core-web/src/main/java/org/silverpeas/core/web/mvc/controller/AbstractAdminComponentSessionController.java:172]: `userManageableSpaceIdSet` contains the identifier of spaces the user can manage.
For example, if the user is set explicitly on `Space A` `userManageableSpaceIdSet` contains `Space A` identifier and those of `Space A` children.
In read only case, parent space of `Space A` must be accessed by the user without giving him writing rights. So parent spaces of spaces contained into `userManageableSpaceIdSet` have to be verified.
Each space path of space indentifiers included into the set MUST be verified. For performances, If the parent identifier of space identifier of the set is already itself in the set, there is no need to check the path as it has been verified on the previous if clause.

**Review by mmoqui — APPROVED** [review]

**Pull request merged**